### ISSUE TITLE
`globstar` option of `expandGlob{,Sync}` is ignored

### ISSUE THREAD

**Issue by kidonng** [denoland/deno_std#2571]
**Steps to Reproduce**

```ts
import {expandGlobSync} from 'https://deno.land/std@0.153.0/fs/mod.ts'
import {assertEquals} from 'https://deno.land/std@0.153.0/testing/asserts.ts'

const prepare = () => {
	const dir = Deno.makeTempDirSync()
	for (const item of ['foo', 'foo/nested', 'bar', 'bar/nested']) {
		Deno.mkdirSync(`${dir}/${item}`, {recursive: true})
	}

	return dir
}

Deno.test('globstar true', () => {
	const dir = prepare()

	assertEquals(
		[
			...expandGlobSync(`${dir}/**`, {
				globstar: true,
			}),
		].length,
		5,
	)

	Deno.removeSync(dir, {recursive: true})
})

Deno.test('globstar false', () => {
	const dir = prepare()

	assertEquals(
		[
			...expandGlobSync(`${dir}/**`, {
				globstar: false,
			}),
		].length,
		2,
	)

	Deno.removeSync(dir, {recursive: true})
})
```

**Expected behavior**

Tests pass

**Describe the bug**

```
running 2 tests from ./test.ts
globstar true ... ok (8ms)
globstar false ... FAILED (6ms)

 ERRORS 

globstar false => ./test.ts:28:6
error: AssertionError: Values are not equal:


    [Diff] Actual / Expected


-   5
+   2

  throw new AssertionError(message);
        ^
    at assertEquals (https://deno.land/std@0.153.0/testing/asserts.ts:183:9)
    at file:///workspaces/infra/test.ts:31:2

 FAILURES 

globstar false => ./test.ts:28:6

FAILED | 1 passed | 1 failed (46ms)

error: Test failed
```

**Environment**

- OS: Ubuntu 18.04
- deno version: deno 1.25.0
- std version: 0.153.0

**Comment by kidonng** [comment]
Slightly related, why does `**` globs the parent itself? This is different from shell behavior and in my opinion harmful.

**Comment by nayeemrmn** [comment]
> Slightly related, why does `**` globs the parent itself? This is different from shell behavior and in my opinion harmful.

Hmm, it seems the conventional behaviour is to match the parent when there is a prefix before the globstar but not if it's on its own:
```
$ echo examples/**
examples/ examples/README.md examples/cat.ts examples/cat_test.ts examples/catj.ts examples/catj_test.ts examples/chat examples/chat/index.html examples/chat/server.ts examples/chat/server_test.ts examples/colors.ts examples/colors_test.ts examples/curl.ts examples/curl_test.ts examples/echo_server.ts examples/echo_server_test.ts examples/flags.ts examples/gist.ts examples/test.ts examples/test_test.ts examples/testdata examples/testdata/cat examples/testdata/cat/hello.txt examples/testdata/cat/world.txt examples/testdata/catj examples/testdata/catj/array.json examples/testdata/catj/object.json examples/testdata/catj/simple-array.json examples/testdata/catj/simple-object.json examples/welcome.ts examples/welcome_test.ts examples/xeval.ts examples/xeval_test.ts
$ cd examples; echo **
README.md cat.ts cat_test.ts catj.ts catj_test.ts chat chat/index.html chat/server.ts chat/server_test.ts colors.ts colors_test.ts curl.ts curl_test.ts echo_server.ts echo_server_test.ts flags.ts gist.ts test.ts test_test.ts testdata testdata/cat testdata/cat/hello.txt testdata/cat/world.txt testdata/catj testdata/catj/array.json testdata/catj/object.json testdata/catj/simple-array.json testdata/catj/simple-object.json welcome.ts welcome_test.ts xeval.ts xeval_test.ts
```
we should fix that as well I suppose.

**Comment by kidonng** [comment]
> Hmm, it seems the conventional behaviour is to match the parent when there is a prefix before the globstar

Thanks, I was thinking about globbing in Fish 😅

**Comment by nkabbara** [comment]
Hi @kidonng! I think this MR should take care of it? Please let me know if you think I should handle differently.

As far as `**` returning the parent, I went back and forth a couple of times, but in the end decided that maybe it should be included. My reasoning:
```
nkabbara@/tmp
$ tree a
a
└── b
    ├── c
    └── d

3 directories, 0 files
nkabbara@/tmp
$ echo /tmp/a/**
/tmp/a/ /tmp/a/b /tmp/a/b/c /tmp/a/b/d
nkabbara@/tmp
```
expandGlob is always getting the absolute path of a root. So when a `**` is present, I think including parent seem to mimic what bash is doing above.

**Comment by kt3k** [comment]
closed by #2779
